ios/browser/api/ads/brave_ads.mm

```diff
@@ -1311,46 +1328,27 @@ - (void)getSiteHistory:(int)max_count
 
 - (void)UrlRequest:(brave_ads::mojom::UrlRequestInfoPtr)url_request
           callback:(brave_ads::UrlRequestCallback)callback {
-  std::map<brave_ads::mojom::UrlRequestMethodType, std::string> methodMap{
-      {brave_ads::mojom::UrlRequestMethodType::kGet, "GET"},
-      {brave_ads::mojom::UrlRequestMethodType::kPost, "POST"},
-      {brave_ads::mojom::UrlRequestMethodType::kPut, "PUT"}};
-
-  const auto copiedURL = url_request->url;
-
-  auto cb = std::make_shared<decltype(callback)>(std::move(callback));
-  const auto __weak weakSelf = self;
-  return [self.commonOps
-      loadURLRequest:url_request->url.spec()
-             headers:url_request->headers
-             content:url_request->content
-        content_type:url_request->content_type
-              method:methodMap[url_request->method]
-            callback:^(
-                const std::string& errorDescription, int statusCode,
-                NSData* responseData,
-                const base::flat_map<std::string, std::string>& headers) {
-              const auto strongSelf = weakSelf;
-              if (!strongSelf || ![strongSelf isServiceRunning]) {
-                return;
-              }
-
-              std::string response;
-              if (responseData && responseData.length > 0) {
-                response =
-                    std::string(static_cast<const char*>(responseData.bytes),
-                                responseData.length);
-              }
+  CHECK(url_request);
+  if (!networkClient) {
+    brave_ads::mojom::UrlResponseInfo failed_mojom_url_response;
+    failed_mojom_url_response.code = net::ERR_FAILED;
+    return std::move(callback).Run(failed_mojom_url_response);
+  }
 
-              brave_ads::mojom::UrlResponseInfo mojom_url_response;
-              mojom_url_response.url = copiedURL;
-              mojom_url_response.code = statusCode;
-              mojom_url_response.body = response;
-              mojom_url_response.headers = headers;
-              if (cb) {
-                std::move(*cb).Run(mojom_url_response);
-              }
-            }];
+  networkClient->SendRequest(
+      std::move(url_request),
+      base::BindOnce(
+          [](brave_ads::UrlRequestCallback callback,
+             brave_ads::mojom::UrlResponseInfoPtr mojom_url_response) {
+            if (!mojom_url_response) {
+              brave_ads::mojom::UrlResponseInfo failed_mojom_url_response;
+              failed_mojom_url_response.code = net::ERR_FAILED;
+              return std::move(callback).Run(failed_mojom_url_response);
```

Comment:
nit: return on a separate line after the callback run

Reply:
I know it returns void, inconsistent in this case is fine and I think the rest should be fixed up in the future but realistically `brave_ads.mm` should go away entirely in the future anyways